packages/dd-trace/src/runtime_metrics.js

```diff
@@ -202,6 +222,29 @@ function captureHeapSpace () {
     client.gauge('runtime.node.heap.physical_size.by.space', stats[i].physical_space_size, tags)
   }
 }
+function captureGCMetrics () {
+  if (!gcProfiler) return
+
+  const profile = gcProfiler.stop()
```

Comment:
Interesting. I'm currently using `PerformanceObserver` for gathering GC events for profiler timeline view, I'll look into whether it'd make sense to switch to `GCProfiler` too. Presumably we can have multiple instances of the `GCProfiler` running?

Reply:
Yes it should be possible to run multiple instances, although maybe we could share one somehow. Not sure how beneficial that would be though. The main advantage to GCProfiler, although I have not benchmarked this directly, is that it only reports the data once when you stop it instead of potentially hundreds or thousands of times per second under high load. However, it does buffer for longer than `PerformanceObserver`, but it does so on the native side. I added it to avoid a lot of back and forth from JS to native as it seemed like a better trade-off.

Reply:
Interesting – thank you, I'll keep this in mind.